File: thousands-tv/wildcard/packages/interfaces/src/db/interfaces/stream.ts
import { Schema, Types } from "mongoose";

export enum ChatApp {
    WILDCARD = "wildcard",
    STREMECOIN = "stremecoin",
    NONE = "none",
    BOOST = "boost",
}
export interface IStream {
    serverId: Schema.Types.ObjectId;
    stageId: Schema.Types.ObjectId;
    vendorEventId: string;
    name: string;
    description: string;
    status: string;
    stageArn?: string;
    channelArn?: string;
    channelPlaybackUrl?: string;
    chatRoomArn?: string;
    cameraOperatorParticipantToken?: string;
    streamKey: string;
    ingestEndpoint: string;
    chatApp: ChatApp;
    createdAt: Date;
    updatedAt: Date;
    _id: Types.ObjectId;
    __v: number;
}

File: thousands-services/IvsIdleGameShared/Models/Stream.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace IvsIdleGameShared.Models
{
    [BsonIgnoreExtraElements]
    public class EventStream
    {
        [BsonId]
        [BsonElement("_id")]
        public ObjectId Id { get; set; }

        [BsonElement("stageId")]
        public ObjectId? StageId { get; set; }

        [BsonElement("vendorEventId")]
        public string? VendorEventId { get; set; }

        [BsonElement("name")]
        public string? Name { get; set; }

        [BsonElement("description")]
        public string? Description { get; set; }

        [BsonElement("status")]
        public string? Status { get; set; }

        [BsonElement("createdAt")]
        public DateTime? CreatedAt { get; set; }

        [BsonElement("updatedAt")]
        public DateTime? UpdatedAt { get; set; }

        //[BsonIgnore]
        //[BsonElement("__v")]
        //public int __v { get; set; }

        [BsonElement("cameraOperatorParticipantToken")]
        public string? CameraOperatorParticipantToken { get; set; }

        [BsonElement("channelArn")]
        public string? ChannelArn { get; set; }

        [BsonElement("channelPlaybackUrl")]
        public string? ChannelPlaybackUrl { get; set; }

        [BsonElement("chatRoomArn")]
        public string? ChatRoomArn { get; set; }

        [BsonElement("stageArn")]
        public string? StageArn { get; set; }

    }
}
